var/spack/repos/builtin/packages/llvm-amdgpu/package.py

```diff
@@ -34,6 +34,17 @@ class LlvmAmdgpu(CMakePackage):
     variant('build_type', default='Release', values=("Release", "Debug"), description='CMake build type')
     variant('rocm-device-libs', default=True, description='Build ROCm device libs as external LLVM project instead of a standalone spack package.')
     variant('openmp', default=True, description='Enable OpenMP')
+    variant(
+        "llvm_dylib",
+        default=False,
+        description="Build LLVM shared library, containing all "
+        "components in a single shared library",
+    )
+
+    # TODO: Version mapping needs more detail
+    provides('libllvm@:12', when='@3.5:4.2')
+    provides('libllvm@12', when='@4.2:4.3')
+    provides('libllvm@13', when='@4.3:')
```

Comment:
Does it provide the appropriate library when llvm_dylib is False?

Reply:
Oops, I messed up the patch on this, it used to have a
```python
provides('libllvm +llvm_dylib', when='+llvm_dylib')
```